### PR TITLE
chore(sqlserverflex): Add nil pointer checks and tests for the output…

### PR DESCRIPTION
**Pull request by Fyusel**
…Result functions

### CODE REVIEW

File: internal/cmd/beta/sqlserverflex/database/describe/describe_test.go [hunk 1]
```diff
@@ -237,3 +236,37 @@ func TestBuildRequest(t *testing.T) {
...
+			args:    args{},
+			wantErr: true,
+		},
+		{
+			name: "only database as argument",
+			args: args{
+				resp: &sqlserverflex.GetDatabaseResponse{Database: &sqlserverflex.SingleDatabase{}},
+			},
+			wantErr: false,
+		},
```
Comment (rubenhoenle) [inline internal/cmd/beta/sqlserverflex/database/describe/describe_test.go:261]:
```go
{
	name: "empty response",
	args: args{
		resp: &sqlserverflex.GetDatabaseResponse{},
	},
	wantErr: false,
},
```

I think this additional test case can't curt

File: internal/cmd/beta/sqlserverflex/database/list/list_test.go [hunk 2]
```diff
@@ -210,3 +209,30 @@ func TestBuildRequest(t *testing.T) {
...
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name:    "empty",
+			args:    args{},
+			wantErr: false,
+		},
```
Comment (rubenhoenle) [inline internal/cmd/beta/sqlserverflex/database/list/list_test.go:227]:
```go
{
	name:    "empty database in databases slice",
	args:    args{
            databases: []sqlserverflex.Database{{}},
        },
	wantErr: false,
},
```

File: internal/cmd/beta/sqlserverflex/instance/list/list_test.go [hunk 3]
```diff
@@ -186,3 +185,30 @@ func TestBuildRequest(t *testing.T) {
...
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name:    "empty",
+			args:    args{},
+			wantErr: false,
+		},
```
Comment (rubenhoenle) [inline internal/cmd/beta/sqlserverflex/instance/list/list_test.go:203]:
```go
{
	name:    "empty instance in instances slice",
	args:    args{
            instances: []sqlserverflex.InstanceListInstance{{}},
        },
	wantErr: false,
},
```

File: internal/cmd/beta/sqlserverflex/user/list/list_test.go [hunk 4]
```diff
@@ -203,3 +202,30 @@ func TestBuildRequest(t *testing.T) {
...
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name:    "empty",
+			args:    args{},
+			wantErr: false,
```
Comment (rubenhoenle) [inline internal/cmd/beta/sqlserverflex/user/list/list_test.go:219]: also here an additional test case could be added with a empty user in the users slice

**Review by rubenhoenle — APPROVED** [review]

**Review by rubenhoenle — APPROVED** [review]
